homeassistant/components/tuya/number.py

```diff
@@ -259,11 +281,11 @@ def __init__(
             # and determine unit of measurement
             if self._status_range.type == "Integer":
                 self._type_data = IntegerTypeData.from_json(self._status_range.values)
-                self._attr_max_value = self._type_data.max
-                self._attr_min_value = self._type_data.min
-                self._attr_step = self._type_data.step
-                if description.unit_of_measurement is None:
-                    self._attr_unit_of_measurement = self._type_data.unit
+                self._attr_max_value = self._type_data.max_scaled
+                self._attr_min_value = self._type_data.min_scaled
+                self._attr_step = self._type_data.step_scaled
```

Comment:
These are bug fixes, it would be nice if we could extract that out of this PR and ship those in a patch release (separate from the new feature).

(including the scaling back of the value)

Reply:
Please explain to me how to do it better. First prepare PR with these bug fixes, wait for merge and then continue with this PR?

Reply:
Yes, if we make a small PR for the bug fixes first (and get that merged), this PR can be git rebased on dev branch after that.

Reply:
Ok, have created PR #59903